lib/queue.js

```diff
@@ -61,6 +63,19 @@ var Queue = function Queue(name, redisPort, redisHost, redisOptions){
     redisOptions.db = redisOpts.DB;
   }
 
+  if (_.isString(redisPort)) {
+    try {
+      var redisUrl = url.parse(redisPort);
+      assert(_.isObject(redisHost) || _.isUndefined(redisHost));
```

Comment:
I don't think there's anything wrong with using asserts in try catch block. It'll produce a nice error if the conditions are not fulfilled and propagate them properly. `assert` is just a syntactic sugar of using an if else block where if the conditions fail you throw an error.

You could argue that instead of throwing an error always saying `Malformed connection string` we should send the error message instead.